core/build.gradle

```diff
@@ -49,66 +37,47 @@ dependencies {
     // Reactive Streams
     compile 'org.reactivestreams:reactive-streams'
 
-    // Internal dependencies which will be shaded by the 'internalDeps' task below.
-    internalDependency 'com.spotify:completable-futures'
-    internalDependency 'it.unimi.dsi:fastutil'
-    internalDependency 'org.reflections:reflections'
-
-    // Used by the 'relocateInternalDependencies', because ShadowTask doesn't let me specify a file.
-    trimmedInternalDependency files(internalDependenciesJarTrimmed)
+    // Reflections
+    compile 'org.reflections:reflections'
 }
 
-task trimInternalDependencies(type: ProGuardTask) {
-    injars files(configurations.internalDependency.collect()), filter: '!META-INF/**'
-    outjars internalDependenciesJarTrimmed
-    libraryjars files(configurations.runtime.collect() - configurations.internalDependency.collect())
+tasks.shadedJar.archiveName = "${tasks.jar.baseName}-untrimmed-${tasks.jar.version}.jar"
+
+task trimShadedJar(type: ProGuardTask) {
+    def shadedFile = tasks.shadedJar.archivePath
+    def shadedAndTrimmedFile = file(shadedFile.path.replaceFirst('-untrimmed-', '-shaded-'))
+    injars shadedFile
+    outjars shadedAndTrimmedFile
+    libraryjars files(configurations.runtime.collect())
     libraryjars file("${System.getProperty('java.home')}/lib/rt.jar")
 
     dontoptimize
     dontobfuscate
     dontwarn // Ignore the harmless 'missing classes' warnings related with the optional dependencies.
 
     keepattributes 'Signature, InnerClasses, Annotation'
+    keep 'class !com.linecorp.armeria.internal.**,com.linecorp.armeria.** { *; }'
 
     // CompletableFutures
-    keepclasses 'com.spotify.futures.CompletableFutures'
+    keepclasses 'com.linecorp.armeria.internal.futures.CompletableFutures'
 
     // FastUtil
-    keepinterfaces 'it.unimi.dsi.fastutil.ints.Int2ObjectMap',
-                   'it.unimi.dsi.fastutil.ints.Int2ObjectFunction'
-    keepclasses 'it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap',
-                'it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap'
+    keepclasses 'com.linecorp.armeria.internal.fastutil.ints.Int2ObjectMap',
```

Comment:
You can probably remove most or all of these as with access to the source files, proguard can find these automatically.

At least you don't need these interfaces I think since they actually are ignored by proguard if I understand correctly (see other comment).